### DISCUSSION TITLE
Observers in sequel

### DISCUSSION THREAD

**Discussion by mybooking-es** [jeremyevans/sequel#1999] [General]
We are planning to migrate our application from DataMapper to Sequel. I haven't found a way to implement observers. Is it possible in Sequel?

Thanks

**Comment by jeremyevans** [comment]
What actions/behavior do you want to observe?

**Comment by mybooking-es** [comment]
Thank you for your answer.  

We need to observe after_update on a concrete model. We are actually doing it in a custom class including DataMapper::Observer and observe a registered Model. We have different observers on the same model depending on the relationships.

**Reply by jeremyevans** [reply]
Seems like something like the following would work:

```ruby
class ModelToObserve < Sequel::Model
  def after_update
    # your code
    super
  end
end
```

**Reply by mybooking-es** [reply]
Thank you. Could even built on a separate Module and be included? Our models are a little big and we try to separate some logic this way.

**Reply by jeremyevans** [reply]
Definitely:

```ruby
module MyObserver
  def after_update
    # your code
    super
  end
end

ModelToObserve.include MyObserver
```
